tests/test_dse.py

```diff
@@ -139,12 +139,12 @@ def test_tti_rewrite_aggressive(tti_nodse):
     # * four Arrays are allocated on the heap
     # * two Arrays are allocated on the stack and only appear within an efunc
     arrays = [i for i in FindSymbols().visit(op) if i.is_Array]
-    assert len(arrays) == 4
+    assert len(arrays) == 3
```

Comment:
I'm confused, can't see where this change is coming from.

Reply:
Optimization in TTI. We now only need 3 temporaries for the time invariants fields , instead of 4 . Becuase instead of storing:
* sin(theta)
* cos(theta)
* sin(phi)
* cos(phi)

the DSE now detects
* sin(theta)
* cos(theta)*sin(phi)
* sin(theta)*cos(phi)

 This is quite cute. Go check out the generated code

Reply:
Nice